### ISSUE TITLE
BUG/API: Can't use DataFrame.index string methods

### ISSUE THREAD

**Issue by TomAugspurger** [dask/dask#1560]
```python
import pandas as pd
import dask.dataframe as dd

df = pd.DataFrame({'a': [1, 2]}, index=['a', 'b'])
a = dd.from_pandas(df, npartitions=2)
a.index.str.lower()
 ```

raises with

```pytb
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-9-b8d2a0689fbe> in <module>()
      1 df = pd.DataFrame({'a': [1, 2]}, index=['a', 'b'])
      2 a = dd.from_pandas(df, npartitions=2)
----> 3 a.index.str.lower()

/Users/tom.augspurger/Envs/py3/lib/python3.5/site-packages/dask/dask/dataframe/core.py in _function_map(self, key, *args)
   3184
   3185     def _function_map(self, key, *args):
-> 3186         out = self.call(self._series._meta, key, *args)
   3187         meta = pd.Series([], dtype=out.dtype, name=getattr(out, 'name', None))
   3188         return map_partitions(self.call, self._series, key, *args, meta=meta)

/Users/tom.augspurger/Envs/py3/lib/python3.5/site-packages/dask/dask/dataframe/core.py in call(obj, attr, *args)
   3237     @staticmethod
   3238     def call(obj, attr, *args):
-> 3239         return getattr(obj.str, attr)(*args)
   3240
   3241

/Users/tom.augspurger/Envs/py3/lib/python3.5/site-packages/pandas-0.19.0rc1+21.ge596cbf-py3.5-macosx-10.11-x86_64.egg/pandas/core/base.py in __get__(self, instance, owner)
    239             # this ensures that Series.str.<method> is well defined
    240             return self.accessor_cls
--> 241         return self.construct_accessor(instance)
    242
    243     def __set__(self, instance, value):

/Users/tom.augspurger/Envs/py3/lib/python3.5/site-packages/pandas-0.19.0rc1+21.ge596cbf-py3.5-macosx-10.11-x86_64.egg/pandas/core/strings.py in _make_str_accessor(self)
   1851                            "(i.e. inferred_type is 'string', 'unicode' or "
   1852                            "'mixed')")
-> 1853                 raise AttributeError(message)
   1854             if self.nlevels > 1:
   1855                 message = ("Can only use .str accessor with Index, not "

AttributeError: Can only use .str accessor with string values (i.e. inferred_type is 'string', 'unicode' or 'mixed')
```

Just a guess, but the inferred type of `a._meta.index` being empty messes things up when we get to pandas land. Not time to look for a fix, so just the bug report for now 😄

**Comment by jcrist** [comment]
Thanks for the bug report, you're right about the issue with empty series doing funny things. Fixing now.

**Comment by jcrist** [comment]
Fixed in #1561.
